### ISSUE TITLE
Block state / Material access

### ISSUE THREAD

**Issue by davidmayr** [olijeffers0n/PatheticAPI#43]
### Is your feature request related to a problem?

It's not really possible to check for block states when checking for blocks in the pathfinding strategy (e.g. pathfind only on trapdoors if they are closed)

### Describe the solution you'd like.

Having access to the block state or at least the material of the block in PathBlock

### Describe alternatives you've considered.

Querying the block myself. But I think that would be pretty inefficient. 

### Other

There is probably a reason this doesn't exist. I haven't looked into how pathetic is built, but maybe it is because of performance memory management reasons. However, without this, implementing more complex strategies is not really possible.

**Comment by bsommerfeld** [comment]
thank you for the suggestion. i'll look into it.

**Comment by olijeffers0n** [comment]
Yes, we are working on this. You will likely be able to access the BlockState and Material

**Comment by davidmayr** [comment]
> Yes, we are working on this. You will likely be able to access the BlockState and Material

Awesome. Didn't expect that to happen that quickly. It will definitely allow for more customizable pathfinding. Since that is why I chose pathetic in the first place.  (Vanilla Pathfinding chose the most stupid paths and doesn't have a strategy api)

**Comment by bsommerfeld** [comment]
we now have a working, probably not yet final, version of this feature in our stage branch. not sure when we will release this, but you can already use it by using the stage as your dependency version.

```
        <dependency>
	    <groupId>com.github.patheloper.pathetic</groupId>
	    <artifactId>pathetic-mapping</artifactId>
	    <version>stage-9446861f8c-1</version>
	</dependency>
```
PathBlock: https://github.com/patheloper/pathetic/blob/stage/pathetic-api/src/main/java/org/patheloper/api/wrapper/PathBlock.java#L19

please note, that due to incompatibility, the blockstate is a spigot v1.13+ only feature. older versions will only grant the material.

**Comment by bsommerfeld** [comment]
we've overhauled and improved this feature so my message before is obsolete.
this is most likely the final version which will also be released.

version:
```
	<dependency>
	    <groupId>com.github.patheloper.pathetic</groupId>
	    <artifactId>pathetic-api</artifactId>
	    <version>stage-08b41453d1-1</version>
	</dependency>
```

fyi: there is now a [BlockInformation](https://github.com/patheloper/pathetic/blob/stage/pathetic-api/src/main/java/org/patheloper/api/wrapper/BlockInformation.java) in PathBlock which contains the Material and BlockState

**Comment by bsommerfeld** [comment]
this feature is now officially available with [version 2.2.1](https://github.com/patheloper/pathetic/releases/tag/2.2.1)
